### PR TITLE
File explorer

### PR DESCRIPTION
**Pull request by ioedeveloper**

### CODE REVIEW

**Comment by ryestew** [comment]
When I try to connect with remixd, with the absolute path to a folder, in the file explorer, I get the **localhost** folder at the root of my computer.

**Comment by ryestew** [comment]
the 4th icon - 
image
is confusing me - I think the text instead of:
Upload Local files to the Browser Storage Explorer

should be:
Load a local file into Remix's browser folder

**Comment by ryestew** [comment]
So its probably beyond the scope of what you are doing for this 1st move to React - but it would be nice if a user could drag a file into and out of a folder.  Its a currently a bit hard to clean up the folders & files ( in master and in this branch).

**Comment by ryestew** [comment]
Creating a file and folder in localhost is great!

**Review by ryestew — COMMENTED** [review]
the folder icons are too dark in the light theme.  I put filter: invert(0.5); and it looks like this now:

image

**Comment by ioedeveloper** [comment]
> When I try to connect with remixd, with the absolute path to a folder, in the file explorer, I get the **localhost** folder at the root of my computer.

I also experienced this bug after I rebased to master. It is somehow related to @yann300 PR https://github.com/ethereum/remix-project/pull/689/files.

File: apps/remix-ide/src/app/panels/file-panel.js [hunk 1]
```diff
@@ -44,63 +47,86 @@ const profile = {
...
-    self._deps.fileProviders.localhost.event.register('connecting', (event) => {
+    this.event = new EventManager()
+    // fileExplorer.ensureRoot()
+    this._deps.fileProviders.localhost.event.register('connecting', (event) => {
     })
 
-    self._deps.fileProviders.localhost.event.register('connected', (event) => {
-      fileSystemExplorer.show()
-    })
+    // this._deps.fileProviders.localhost.event.register('connected', (event) => {
```
Comment (yann300) [inline apps/remix-ide/src/app/panels/file-panel.js:81]: This is handled somewhere else or you don't want anymore these event to apply?

Comment (ioedeveloper) [reply apps/remix-ide/src/app/panels/file-panel.js:81]: They are still valid events. I will update my PR.

**Comment by bunsenstraat** [comment]
If I right click on the icons, not on the names, I don't get a context menu, is that by design?

**Review by yann300 — APPROVED** [review]

**Comment by ioedeveloper** [comment]
> If I right click on the icons, not on the names, I don't get a context menu, is that by design?

Yes it is. I had trouble retreiving filename if icon is clicked so I limited it to the text field.

**Pull request merged**